client/app/telecom/telephony/line/calls/simultaneousLines/telecom-telephony-line-calls-simultaneousLines.controller.js

```diff
@@ -118,43 +72,90 @@ angular.module("managerApp")
             self.options.simultaneousLines = self.saved.simultaneousLines;
         };
 
-        self.save = debounce(function () {
-
-            self.contracts = null;
+        self.save = function () {
             self.prices = null;
             self.contractsAccepted = false;
+            self.showDoRemoveButtons = false;
 
             if (self.needSave()) {
-
                 if (self.saved.simultaneousLines < self.options.simultaneousLines) {
-                    self.loading.save = true;
                     self.showDoRemoveButtons = false;
-                    return self.showOrder().finally(function () {
-                        self.loading.save = false;
-                    });
+                    self.recalculatePrices();
                 }
-                if (self.saved.simultaneousLines > self.options.simultaneousLines) {
+                if (self.saved.simultaneousLines > self.options.simultaneousLines && self.options.simultaneousLines) {
                     self.showDoRemoveButtons = true;
                 }
-
-
             } else {
                 self.showDoRemoveButtons = false;
             }
+        };
+
+        self.recalculatePrices = function () {
+            var quantity = self.options.simultaneousLines - self.saved.simultaneousLines;
 
-            return $q.when(null);
-        }, 600, false);
+            self.prices = {
+                tax: { text: $filter("currency")(unitPrices.tax.value * quantity) },
+                withTax: { text: $filter("currency")(unitPrices.withTax.value * quantity) },
+                withoutTax: { text: $filter("currency")(unitPrices.withoutTax.value * quantity) }
```

Comment:
are you sure that prices are prices are rounded correctly ?

Reply:
In general, the result of multiplication unfiltered have more than two decimals, the lack of precision belongs in those 2n+ decimals, so I suggest that, with the filter applied, it may not be problematic

Reply:
ok !